src/bvals/cc/sparse_flux_correction_cc_in_one.cpp

```diff
@@ -0,0 +1,335 @@
+        if (nb.fid == BoundaryFace::inner_x1 || nb.fid == BoundaryFace::outer_x1) {
+          dir = X1DIR;
+          if (nb.fid == BoundaryFace::inner_x1)
+            ie = is;
+          else
+            is = ++ie;
+          if (nb.ni.fi1 == 0)
+            je -= pmb->block_size.nx2 / 2;
+          else
+            js += pmb->block_size.nx2 / 2;
+          if (nb.ni.fi2 == 0)
+            ke -= pmb->block_size.nx3 / 2;
+          else
+            ks += pmb->block_size.nx3 / 2;
+        } else if (nb.fid == BoundaryFace::inner_x2 || nb.fid == BoundaryFace::outer_x2) {
+          dir = X2DIR;
+          if (nb.fid == BoundaryFace::inner_x2)
+            je = js;
+          else
+            js = ++je;
+          if (nb.ni.fi1 == 0)
+            ie -= pmb->block_size.nx1 / 2;
+          else
+            is += pmb->block_size.nx1 / 2;
+          if (nb.ni.fi2 == 0)
+            ke -= pmb->block_size.nx3 / 2;
+          else
+            ks += pmb->block_size.nx3 / 2;
+        } else if (nb.fid == BoundaryFace::inner_x3 || nb.fid == BoundaryFace::outer_x3) {
+          dir = X3DIR;
+          if (nb.fid == BoundaryFace::inner_x3)
+            ke = ks;
+          else
+            ks = ++ke;
+          if (nb.ni.fi1 == 0)
+            ie -= pmb->block_size.nx1 / 2;
+          else
+            is += pmb->block_size.nx1 / 2;
+          if (nb.ni.fi2 == 0)
+            je -= pmb->block_size.nx2 / 2;
+          else
+            js += pmb->block_size.nx2 / 2;
+        } else {
+          PARTHENON_FAIL("Flux corrections only occur on faces for CC variables.");
+        }
+
+        auto &flx = v->flux[dir];
+        buf_pool_t<Real>::weak_t &buf_arr = buf.buffer();
+        const int nl = flx.GetDim(6);
+        const int nm = flx.GetDim(5);
+        const int nn = flx.GetDim(4);
+        const int nk = ke - ks + 1;
+        const int nj = je - js + 1;
+        const int ni = ie - is + 1;
+        const int NjNi = nj * ni;
+        const int NkNjNi = nk * NjNi;
+        const int NnNkNjNi = nn * NkNjNi;
+        const int NmNnNkNjNi = nm * NnNkNjNi;
+        if (nl * NmNnNkNjNi > buf_arr.size()) {
+          PARTHENON_FAIL("Buffer to small")
+        }
+
+        Kokkos::parallel_for(
+            "SendFluxCorrection",
+            Kokkos::RangePolicy<>(parthenon::DevExecSpace(), 0, nl * NmNnNkNjNi),
+            KOKKOS_LAMBDA(const int loop_idx) {
+              const int l = loop_idx / NmNnNkNjNi;
+              const int m = (loop_idx % NmNnNkNjNi) / NnNkNjNi;
+              const int n = (loop_idx % NnNkNjNi) / NkNjNi;
+              const int k = (loop_idx % NkNjNi) / NjNi + ks;
+              const int j = (loop_idx % NjNi) / ni + js;
+              const int i = loop_idx % ni + is;
+
+              const int idx =
+                  i - is + ni * (j - js + nj * (k - ks + nk * (n + nn * (m + nm * l))));
+              flx(l, m, n, k, j, i) = buf_arr(idx);
+            });
+#ifdef MPI_PARALLEL
+        Kokkos::fence();
+#endif
```

Comment:
I don't think we need a fence here, do we?

Reply:
Yeah, I think that was a paranoia fence.

Reply:
gone.